packages/react-instantsearch-core/src/connectors/__tests__/connectInfiniteHits.js

```diff
@@ -83,10 +138,63 @@ describe('connectInfiniteHits', () => {
           page: 2,
           hitsPerPage: 8,
           nbPages: 10,
+          queryID: 'theQueryID_2',
         },
       });
 
-      expect(res3.hits).toEqual([...hits, ...hits2, ...hits3]);
+      expect(res3.hits).toEqual(
+        [...hits, ...hits2, ...hits3].map(hit => expect.objectContaining(hit))
+      );
+      expect(res3.hits.map(hit => hit.__position)).toEqual([
+        // page: 0, hitsPerPage: 6
+        1,
+        2,
+        3,
+        4,
+        5,
+        6,
+        // page: 1, hitsPerPage: 6
+        7,
+        8,
+        9,
+        10,
+        11,
+        12,
+        // page: 2, hitsPerPage: 8
```

Comment:
I'd clarify after this that it's an existing limitation of the engine of not being able to correctly compute an offset

Reply:
yes, I'll add the following
> hitsPerPage changed from 6 to 8, elements 13-16 are skipped

Reply:
fixed here: https://github.com/algolia/react-instantsearch/pull/2217/commits/45220e54ada5f9734405e7b0d971dc6f238243ee